asserts/ifacedecls.go

```diff
@@ -78,6 +85,12 @@ func compileAttrMatcher(cc compileContext, constraints interface{}) (attrMatcher
 		if !cc.hadMap {
 			return nil, fmt.Errorf("first level of non alternative constraints must be a set of key-value contraints")
 		}
+		if strings.HasPrefix(x, "$") {
+			if x == "$MISSING" {
+				return missingAttrMatcher{}, nil
+			}
+			return compileEvalAttrMatcher(cc, x)
+		}
```

Comment:
Logic here gives a pretty good feeling. Small spot clearly returning the alternatives for handling different rule styles.